src/Foundation/NSObject2.cs

```diff
@@ -1045,6 +1046,34 @@ public IDisposable AddObserver (NSString key, NSKeyValueObservingOptions options
 			AddObserver (o, key, options, o.Handle);
 			return o;
 		}
+
+		public static NSObject Alloc (Class kls)
+		{
+			var h = Messaging.IntPtr_objc_msgSend (kls.Handle, Selector.GetHandle (Selector.Alloc));
+			return new NSObject (h, true);
+		}
+
+		public void Init ()
+		{
+			if (handle == IntPtr.Zero)
+				throw new Exception ("you have not allocated the native object");
+
+			handle = Messaging.IntPtr_objc_msgSend (handle, Selector.GetHandle ("init"));
+		}
+
+		public static void InvokeInBackground (Action action)
+		{
+			// using the parameterized Thread.Start to avoid capturing
+			// the 'action' parameter (it'll needlessly create an extra
+			// object).
+			new System.Threading.Thread ((v) =>
```

Comment:
The whole create a thread for one callback is worrying. This API might end up in mis-use more often than not.
```suggestion
			new System.Threading.Thread (static (v) =>
```

Reply:
Yeah, it's a somewhat useless and non-optimal API nowadays, but I think we should have the same API on all platforms.

Then we can deal with any problems with the API at the same time for all platforms (separately).